async-nats/src/header.rs

```diff
@@ -339,19 +340,26 @@ impl HeaderValue {
     }
 }
 
-#[derive(Debug, Clone)]
-pub struct ParseHeaderValueError;
```

Comment:
These are intended to be discrete simple errors for `FromStr`, they live outside of the "io-esque" methods on context and client.

This complicates what is a very simple and straight forward discrete error type imo.